test/unit/components/chart/settings.test.js

```diff
@@ -0,0 +1,151 @@
+      expect(console.warn.calledWith('Warning: Required prop `patientData` was not specified in `Settings`.')).to.equal(true);
+      expect(console.warn.calledWith('Warning: Required prop `onClickRefresh` was not specified in `Settings`.')).to.equal(true);
+      expect(console.warn.calledWith('Warning: Required prop `onSwitchToDaily` was not specified in `Settings`.')).to.equal(true);
+      expect(console.warn.calledWith('Warning: Required prop `onSwitchToSettings` was not specified in `Settings`.')).to.equal(true);
+      expect(console.warn.calledWith('Warning: Required prop `trackMetric` was not specified in `Settings`.')).to.equal(true);
+      expect(console.warn.calledWith('Warning: Required prop `uploadUrl` was not specified in `Settings`.')).to.equal(true);
+    });
+
+    it('should render without problems', function () {
+      var props = {
+        bgPrefs: {},
+        chartPrefs: {},
+        patientData: {
+          grouped: { foo: 'bar' }
+        },
+        onClickRefresh: function() {},
+        onSwitchToDaily: function() {},
+        onSwitchToSettings: function() {},
+        onSwitchToWeekly: function() {},
+        trackMetric: function() {},
+        uploadUrl: ''
+      };
+      var settingsElem = React.createElement(Settings, props);
+      var elem = TestUtils.renderIntoDocument(settingsElem);
+      expect(elem).to.be.ok
+    });
+
+    it('should render with missing data message when no grouped data supplied', function () {
+      var props = {
+        bgPrefs: {},
+        chartPrefs: {},
+        patientData: {
+        },
+        onClickRefresh: sinon.spy(),
+        onSwitchToDaily: sinon.spy(),
+        onSwitchToSettings: sinon.spy(),
+        onSwitchToWeekly: sinon.spy(),
+        trackMetric: sinon.spy(),
+        uploadUrl: ''
+      };
+      var settingsElem = React.createElement(Settings, props);
+      var elem = TestUtils.renderIntoDocument(settingsElem);
+      expect(elem).to.be.ok
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+      expect(x.props.children.length).to.equal(3);
+    });
+
+    it('should have a refresh button which should be call onClickRefresh when clicked', function () {
+      var props = {
+        bgPrefs: {},
+        chartPrefs: {},
+        patientData: {
+        },
+        onClickRefresh: sinon.spy(),
+        onSwitchToDaily: sinon.spy(),
+        onSwitchToSettings: sinon.spy(),
+        onSwitchToWeekly: sinon.spy(),
+        trackMetric: sinon.spy(),
+        uploadUrl: ''
+      };
+      var settingsElem = React.createElement(Settings, props);
+      var elem = TestUtils.renderIntoDocument(settingsElem);
+      var refreshButton = TestUtils.findRenderedDOMComponentWithClass(elem, 'btn-refresh');
+
+      expect(props.onClickRefresh.callCount).to.equal(0);
+      TestUtils.Simulate.click(refreshButton);
+      expect(props.onClickRefresh.callCount).to.equal(1);
+    });
+
```

Comment:
Go ahead and just delete this commented-out stuff, if you would, since this kind of testing of the actual settings rendering will eventually be in tideline. Also delete the settings fixture (or even rebase the branch not to include that commit, if you want), please.